apps/condo/domains/organization/sbbol/oauth2.js

```diff
@@ -0,0 +1,170 @@
+            if (this.certificate) {
+                const { pfx, passphrase } = this.certificate
+                withCertificate = { ...withCertificate, https: { pfx, passphrase } }
+            }
+            return withCertificate
+        }
+        // we override standart JWT validation as we do not have JWK from oauth provider
+        client.validateJWT = async (jwt) => {
+            const [ header, payload ] = [jwtDecode(jwt, { header: true }), jwtDecode(jwt)]
+            return { protected: header, payload }
+        }
+        this.client = client
+    }
+
+    createIssuer () {
+        const sbbolIssuer = new Issuer({
+            issuer: this.protectedUrl,
+            authorization_endpoint: this.authUrl,
+            token_endpoint: this.tokenUrl,
+            userinfo_endpoint: this.userInfoUrl,
+            revocation_endpoint: this.revokeUrl,
+        })
+        // turn off JWKS storage as it's not workin with sbbol
+        sbbolIssuer.keystore = async () => null
+        sbbolIssuer.queryKeyStore = async () => null
+        this.issuer = sbbolIssuer
+    }
+
+    authorizationUrlWithParams (checks) {
+        return this.client.authorizationUrl({
+            response_type: 'code',
+            scope: `openid ${this.serviceId}`,
+            ...checks,
+        })
+    }
+
+    async fetchTokens (req, sessionKey) {
+        const params = this.client.callbackParams(req)
+        if (!req.session[sessionKey]) {
+            throw new Error('No check fields in user session')
+        }
+        const { nonce = '', state = '' } = req.session[sessionKey]
+        const tokenSet = await this.client.callback(this.redirectUrl, params, {
+            nonce,
+            state,
+        })
+        return tokenSet
+    }
+
+    async fetchUserInfo (accessToken) {
+        const userInfo = await this.client.userinfo(accessToken)
+        return userInfo
+    }
+
+    async fetchSubscribers (accessToken) {
+        const result = {
+            offers:  await this.client.requestResource(this.offersUrl, accessToken),
+            advanceAcceptances: await this.client.requestResource(this.advanceAcceptancesUrl, accessToken),
+            packageOfServices:  await this.client.requestResource(this.packageOfServicesUrl, accessToken),
+        }
+        return result
+    }
+
+    get userInfoUrl () {
+        return `${this.protectedUrl}/ic/sso/api/v1/oauth/user-info`
+    }
+
+    get tokenUrl () {
+        return `${this.protectedUrl}/ic/sso/api/v2/oauth/token`
+    }
+
+    get authUrl () {
+        return `${this.url}/ic/sso/api/oauth/authorize`
+    }
+
+    get revokeUrl () {
+        return `${this.protectedUrl}/v1/oauth/revoke`
+    }
+
+    get url () {
+        return `${this.host}:${this.port}`
+    }
+
+    get protectedUrl () {
+        return `${this.protectedHost}:${this.port}`
+    }
+    // Subscriptions
+    // Получение сведений о клиентах, подключенных к подпискам и пакетам услуг [GET]
```

Comment:
comments should be in english

Reply:
removed - it was a draft for the next tasks